wgpu-core/src/device/queue.rs

```diff
@@ -1263,7 +1265,9 @@ impl Global {
                                         Some(TextureInner::Native { .. }) => false,
                                         Some(TextureInner::Surface { ref raw, .. }) => {
                                             if raw.is_some() {
-                                                submit_surface_textures_owned.push(texture.clone());
+                                                // Compare the Arcs by pointer as Textures don't implement Eq.
+                                                submit_surface_textures_owned
+                                                    .insert(Arc::as_ptr(&texture), texture.clone());
```

Comment:
Good thought on Arcs, just wondering if we should use the indexes already built into the texture.info.index or something like that.

Reply:
> Good thought on Arcs

I copied what you did in vulkan hal's `Queue::submit` lol

> just wondering if we should use the indexes already built into the texture.info.index

Isn't wgpu trying to remove all those indices and move to just using Arcs?

Reply:
Those indices are staying

Reply:
Let's leave this for now, for the sake of expediency.